File: 938. Range Sum of BST.cpp
class Solution {
public:
    int calculate_sum(TreeNode* root,int low,int high)
    {
        if (root == NULL)
        {
            return 0;
        }
        else if (root->val < low || root->val > high)
        {
            return 0 + calculate_sum(root->left,low,high) + calculate_sum(root->right,low,high);
        }
        else
        {
            return root->val + calculate_sum(root->left,low,high) + calculate_sum(root->right,low,high);
        }
    }
    int rangeSumBST(TreeNode* root, int low, int high) {
        return calculate_sum(root,low,high);
    }
};

File: 938. Range Sum of BST.java
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        if (root == null) return 0;
        return (
           ( root.val >= low && root.val<= high ? root.val:0)+
           (low<root.val ? rangeSumBST(root.left ,low,high):0)+
           (high>root.val ? rangeSumBST(root.right ,low,high):0)
        );
    }
}
